docs/proposals/22-autoscaling-kube-apiserver-via-independent-hpa-and-vpa.md

```diff
@@ -0,0 +1,354 @@
+each recommendation, +2 cores and +2 replicas, resulting in (1 + 2) replicas * (1 + 2) cores. Failing to account for
+the multiplicative interaction between changes, HVPA provides a total of 9 cores to a workload which only needs 5. 
+
+### Other Benefits to Replacing the Existing Solution
+HVPA applies recommendations via direct edits to the controlled object (e.g. `Deployment`). This approach requires
+nonobvious special handling by Gardener code outside HVPA, which is error-prone.
+
+### Goals
+- Improve ShootKapi autoscaling stability by eliminating the stability issues existing in the current HVPA-based solution.
+- Improve ShootKapi operational efficiency at low application load levels, by avoiding excessive replicas driven by
+  HVPA's policy for early horizontal scale-out.
+- Allow usage of VPA for right-sizing the resource requests during the entire time and not only under certain conditions.
+- Enable future migration away from HVPA by removing all dependency on it for scaling ShootKapi.
+- Reduce Gardener custom code base and ongoing maintenance effort.
+
+### Non-Goals
+There are other components currently scaled via HVPA. Changing the autoscaling mode of these, and of `kube-apiserver`s
+other than the one in the shoot control plane, is outside the scope of this GEP. These workloads are subject to a
+different mode of autoscaling (e.g. vertical-only, where HVPA is only used to stabilize VPA behavior), and it is
+advantageous from project execution perspective to address them as a separate concern.
+
+## Proposal
+ShootKapi HVPA is replaced by individual HPA and VPA. Undesirable interference between HPA and VPA is avoided through
+the use of sufficiently independent driving signals. HPA is driven by the rate of HTTP requests to kube-apiserver. VPA is
+driven by CPU and memory usage metrics.
+
+### Rationale
+A rough, heuristic horizontal sizing sets the stage for optimal VPA operation (horizontal size is within stable and
+reasonably efficient range). Vertical scaling fine-tunes requests, ensuring high efficiency (requests not too high)
+and stability (requests not too low).
+
+### Design Outline
+One HPA and one VPA resources are deployed alongside each ShootKapi deployment object. The VPA controls CPU and memory
+requests but not limits. The HPA is driven by the average rate of API requests per pod of the target kube-apiserver
+deployment.
+
+A new component named gardener-custom-metrics is deployed on each seed. It directly scrapes metrics data from all
+ShootKapi pods on the seed and derives custom metrics based on it. The component is registered as an extension API
+service to the seed kube-apiserver. It occupies the custom metrics API extension point and is responsible for providing
+all custom metrics for the seed kube-apiserver, including the one driving the ShootKapi HPA.
+
+![Design outline](./assets/gep22-design-outline.png "Design outline")
```

Comment:
Nit: there is no "Soil" in gardener

Reply:
I think at some point we added it to the glossary: https://gardener.cloud/docs/glossary/

Reply:
...but indeed it is not actually necessary that the seed control plane be hosted by a soil cluster. I'll replace "soil" with something vague. I wish we had a good general term for this...

Reply:
I replaced "Soil" with "Seed control plane host". Let me know in case you come up with a better name for the diagram box.